chain-signatures/node/src/indexer_eth.rs

```diff
@@ -638,7 +605,48 @@ async fn add_failed_block(
 }
 
 // retry getting latest finalized block from helios with exponential backoff
-async fn fetch_finalized_block_from_helio(
+async fn fetch_block_by_number(
+    helios_client: &Arc<EthereumClient>,
+    block_number: u64,
+    max_retries: u8,
+    base_delay: Duration,
+) -> anyhow::Result<alloy::rpc::types::Block> {
+    let mut retries = 0;
+    loop {
+        match helios_client
+            .get_block(
+                BlockId::Number(BlockNumberOrTag::Number(block_number)),
+                false,
+            )
+            .await
+        {
+            Ok(Some(block)) => return Ok(block),
+            Ok(None) => {
+                let err_msg = "Block number {block_number} not found from Helios client";
+                return Err(anyhow::anyhow!(err_msg));
```

Comment:
just have this function return `Option<Block>` and log inplace. Very easy to end up double logging somewhere this way